dotnet/src/Connectors/Connectors.AI.HuggingFace/TextCompletion/HuggingFaceTextCompletion.cs

```diff
@@ -98,17 +100,24 @@ public HuggingFaceTextCompletion(string apiKey, string model, string endpoint =
     }
 
     /// <inheritdoc/>
-    public async Task<string> CompleteAsync(string text, CompleteRequestSettings requestSettings, CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<ITextCompletionStreamingResult> GetStreamingCompletionsAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        return await this.ExecuteCompleteRequestAsync(text, cancellationToken).ConfigureAwait(false);
+        foreach (var completion in await this.ExecuteGetCompletionsAsync(text, cancellationToken).ConfigureAwait(false))
+        {
+            yield return completion;
+        }
     }
 
-    public IAsyncEnumerable<string> CompleteStreamAsync(
+    /// <inheritdoc/>
+    public async Task<IReadOnlyList<ITextCompletionResult>> GetCompletionsAsync(
```

Comment:
I can't, this works in the other completions because the internal already gives me a IAsyncEnumerable, when using the non async pattern ToAsyncEnumerable actually converts the Task<IReadOnlyList<ItextCompletionStreamingResult>> which breaks the interface.